src/main/java/com/example/learnregex/controllers/RegExOperations.java

```diff
@@ -2,40 +2,79 @@
 
 public class RegExOperations extends Master{
 
     @FXML
-    private TextField textField;
-    @FXML
-    private TextField patternField;
+    private TextField textField, patternField, replaceTextfield, findInput,replaceInput,regexSearch;
     @FXML
-    private Button processBtn;
+    private Button processBtn, replaceButton,findBtn,importBtn,pdfBtn,docsBtn,helpBtn,patternBtn;
     @FXML
-    private TextArea resultArea;
+    private TextArea resultArea,userInput;
     @FXML
-    private TextField replaceTextfield;
+    private Label wordCountLabel;
+
+
+    /**
+     *
+     * @param event
+     */
     @FXML
-    private Button replaceBtn;
+    public void findOperation(ActionEvent event) {
+        if (!userInput.getText().isEmpty() && !findInput.getText().isEmpty()){
+            String result = Master.finder(userInput.getText(),findInput.getText());
+            resultArea.setText(result);
+        }else{
+            ShowAlert.showAlert(
+                    "Error", "Please fill all the fields",
+                    "No input given!!! Please enter the fields and continue! ",
+                    Alert.AlertType.ERROR
+            );
+        }
+
 
+    }
 
+    /**
+     *
+     * @param event
+     */
     @FXML
-    public void basicOperations(ActionEvent event) {
-        String result = Master.operator(textField.getText(),patternField.getText());
-        resultArea.setText(result);
+    public void regexOperation(ActionEvent event) {
+        if (!userInput.getText().isEmpty() && !regexSearch.getText().isEmpty()){
+            String result = Master.finder(userInput.getText(),regexSearch.getText());
+            resultArea.setText(result);
+        }else{
+            ShowAlert.showAlert(
+                    "Error", "Please fill all the fields",
+                    "No input given!!! Please enter the fields and continue! ",
+                    Alert.AlertType.ERROR
+            );
+        }
+
+
     }
 
+    /**
+     * @replaceOperation calls the @Master class and pass the params to perform the replace operation
+     * @param event
+     */
     @FXML
-    public void replaceOperations(ActionEvent event) {
-        String input = textField.getText();
-        String replaceword =  replaceTextfield.getText();
-        String result = input.replaceAll("\\bboy\\b",replaceword);
-        System.out.println(result);
+    public void replaceOperation(ActionEvent event) {
+        if (!userInput.getText().isEmpty() && !findInput.getText().isEmpty() && !replaceInput.getText().isEmpty()){
+        String result = Master.replaceWords(userInput.getText(),findInput.getText(),replaceInput.getText());
+        resultArea.setText(result);
+        }else{
+            ShowAlert.showAlert(
+                    "Error",
+                    "Please fill all the fields",
+                    "No input given!!! Please enter the fields and continue! ",
+                    Alert.AlertType.ERROR);
+        }
     }
+
```

Comment:
`ShowAlert.showAlert()` in all methods is the same(`repetition`), it violates the `DRY `principle, you can use `polymorphism `to handle this.

things to consider
1.since you have a `ShowAlert` class you can define a default `showAlert()` that returns common response message.
2. polymorh the `showAlert()` to accept specific arguments